torch/nn/attention/_flex_attention.py

```diff
@@ -83,6 +83,10 @@ def score_mod(
     """
 
     if torch.compiler.is_dynamo_compiling():
+        # mark head_dim and dim always to be static
+        for x in [query, key, value]:
+            torch._dynamo.mark_static(x, 1)
```

Comment:
imo don't mark `head_dim` as static - no point imo.

Reply:
But we choose the block_size config according the head_dim, if they are dynamic, we can't make the decision during compilation.

Reply:
+1 here, feels bad lol

out of curiosity again here, if the headdim was in fact changing size often enough as to cause automatic dynamic dims what happends here? I imagine it just remains static? And we would ultimately hit the cache limit ? Does the program error when that happends?

Reply:
I think the short answer is we can't get best performance if ```dynamic=True```, because we are going to compile triton kernel with proper block_size based on ```head_dim```. Suppose the ```head_dim``` is 128 during compilation, we would generate best perf triton kernel with block_size X for 128. Then during next iteration, if ```head_dim``` changes to 256, which block_size X is not the best config, but we won't recompile and still use the original kernel, which would lead worse performance.
IMO this is not a big issue, since ```head_dim``` should not be changed across iteration, so I think it doesn't hurt if we make it static.

Reply:
BTW, I was convinced by @Chillee during dinner time, he showed me some cases that needs ```head_dim``` be dynamic, though it could lead worse performance, but we should not mark it as static. I'll update this in a follow-up PR.